jax/_src/lax/lax.py

```diff
@@ -4153,9 +4153,10 @@ def _broadcast_in_dim_typecheck_rule(
 
 def _broadcast_in_dim_transpose_rule(ct, operand, *dyn_shape,
                                      shape, broadcast_dimensions, sharding):
+  aval = core.get_aval(operand)
```

Comment:
Huh, I would've thought operand would be an UndefinedPrimal here. Does get_aval work on that? Why this change?

Reply:
Good point. I changed it because `operand` was sometimes a literal. I thought the implementation was just missing that case (sometimes we assume things are tracers when they can be literals or tracers). But you're right, something else must be going wrong. I'll investigate.

Reply:
Figured it out. I was instantiating zeros inside the linear jaxpr rather than outside it. Gotta dedent your instantiate_zeros.